include/nbl/builtin/hlsl/tgmath.hlsl

```diff
@@ -120,6 +121,26 @@ inline FloatingPoint isinf(NBL_CONST_REF_ARG(FloatingPoint) val)
 #endif
 }
 
+template<typename  T>
+inline T pow(NBL_CONST_REF_ARG(T) x, NBL_CONST_REF_ARG(T) y)
+{
+#ifdef __HLSL_VERSION
+    return spirv::pow(x, y);
+#else
+    return std::pow(x, y);
```

Comment:
pass the template argument explicitly always so you don't fall victim to implicit conversions